engine/gamesys/src/gamesys/scripts/script_buffer.cpp

```diff
@@ -1194,6 +1203,35 @@ namespace dmGameSystem
 
 namespace dmScript
 {
+    LuaHBuffer::LuaHBuffer()
+    {
+    }
+
+    LuaHBuffer::LuaHBuffer(dmBuffer::HBuffer buffer, LuaBufferOwnership ownership)
+    : m_Buffer(buffer)
+    , m_Owner(ownership)
+    {
+    }
+
+    LuaHBuffer::LuaHBuffer(dmResource::HFactory factory, void* buffer_resource)
+    : m_BufferRes(buffer_resource)
+    , m_Owner(OWNER_RES)
+    {
+        if (factory)
+        {
+            m_BufferResVersion = dmResource::GetVersion(factory, buffer_resource);
+            dmResource::GetPath(factory, buffer_resource, &m_BufferResPathHash);
+        }
```

Comment:
Not wure when we create using this constructor, and the factory == 0, but still, it's an OWNER_RES, and we don't keep the factory, so I guess the version should be set to INVALID, so that we don't accidentally use the buffer data?